# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasRenderer.cpp
# Commit: 6f19f43e0a0b
# Full Hash: 6f19f43e0a0bad72fa370191f4cc7f86f58566c2
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 073302d26c5e -r 6f19f43e0a0b gfx/layers/CanvasRenderer.cpp
--- a/gfx/layers/CanvasRenderer.cpp	Thu Jun 11 06:37:38 2020 +0000
+++ b/gfx/layers/CanvasRenderer.cpp	Thu Jun 11 06:37:40 2020 +0000
@@ -46,7 +46,8 @@
   return mData.GetContext() == aData.GetContext();
 }
 
-std::shared_ptr<BorrowedSourceSurface> CanvasRenderer::BorrowSnapshot() const {
+std::shared_ptr<BorrowedSourceSurface> CanvasRenderer::BorrowSnapshot(
+    const bool requireAlphaPremult) const {
   const auto context = mData.GetContext();
   if (!context) return nullptr;
   const auto& provider = context->GetBufferProvider();
@@ -57,7 +58,7 @@
     ss = provider->BorrowSnapshot();
   }
   if (!ss) {
-    ss = context->GetFrontBufferSnapshot();
+    ss = context->GetFrontBufferSnapshot(requireAlphaPremult);
   }
   if (!ss) return nullptr;
 